packages/mediapipe-core/lib/src/ffi_utils.dart

```diff
@@ -76,3 +76,60 @@ int _length(Pointer<Uint8> codeUnits) {
   }
   return length;
 }
+
+/// Converts a pointer to a (representing a list of) floats to a list of
+/// Dart doubles.
+List<double> toDartListDouble(Pointer<Float> floats, {int? length}) {
+  if (floats.isNullPointer) {
+    throw Exception('Unexpected nullptr passed to `toDartListDouble`.');
+  }
+  final codeUnits = floats.cast<Float>();
+  if (length != null) {
+    RangeError.checkNotNegative(length, 'length');
+  } else {
+    length = lengthFloats(codeUnits);
+  }
+  final value = codeUnits.asTypedList(length);
+  return value;
+}
+
+/// Calculates the length of this array of floats by scanning for a value of 0.
+int lengthFloats(Pointer<Float> codeUnits) {
```

Comment:
This code shouldn't be needed as we return the length of the values array.

It also seems rather fragile.

Reply:
True - I added this before realizing the `values_count` field was the flexible length marker. This method is actually no longer used.